lib/webpush/request.rb

```diff
@@ -13,9 +20,14 @@ def perform
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
       req = Net::HTTP::Post.new(uri.request_uri, headers)
       req.body = body
-      http.request(req)
-    rescue => e
-      raise e
+      resp = http.request(req)
+
+      if resp.is_a?(Net::HTTPGone) ||   #Firefox unsubscribed response
+          (resp.is_a?(Net::HTTPBadRequest) && resp.message == "UnauthorizedRegistration")  #Chrome unsubscribed response
+        raise InvalidSubscription.new(resp.inspect)
+      elsif !resp.is_a?(Net::HTTPSuccess)  #unknown/unhandled response error
+        raise ResponseError.new "host: #{uri.host}, #{resp.inspect}\nbody:\n#{resp.body}"
```

Comment:
This code is removes existing behavior by returning `nil` in the "happy path". Previously, we were returning the `response` object. I'd vote for retaining that behavior.